contracts/strategy/ConverterStrategyBase.sol

```diff
@@ -0,0 +1,332 @@
+    __StrategyBase_init(controller_, splitter_);
+    _requireInterface(converter_, InterfaceIds.I_TETU_CONVERTER);
+    tetuConverter = ITetuConverter(converter_);
+    _isReadyToHardWork = true;
+
+  }
+
+  // *************************************************************
+  //                     RESTRICTIONS
+  // *************************************************************
+
+  /// @dev Restrict access only for splitter
+  function _onlyTetuConverter() internal view {
+    require(msg.sender == address(tetuConverter), "CSB: Only TetuConverter");
+  }
+
+  // *************************************************************
+  //                        VIEWS
+  // *************************************************************
+
+  /// @dev See {IERC165-supportsInterface}.
+  function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+    return interfaceId == InterfaceIds.I_CONVERTER_STRATEGY || super.supportsInterface(interfaceId);
+  }
+
+  // *************************************************************
+  //                       OVERRIDES StrategyBase
+  // *************************************************************
+
+  /// @dev Amount of underlying assets converted to pool assets and invested to the pool.
+  function investedAssets() override public view virtual returns (uint) {
+    return _investedAssets;
+  }
+
+  /// @dev Deposit given amount to the pool.
+  function _depositToPool(uint amount) override internal virtual {
+    _doHardWork(false);
+    if (amount == 0) return;
+
+    address _asset = asset;
+    uint assetBalanceBefore = _balance(_asset);
+
+    address[] memory tokens = _depositorPoolAssets();
+    uint len = tokens.length;
+    uint[] memory tokenAmounts = new uint[](len);
+    uint assetAmountForToken = amount / len;
+
+    for (uint i = 0; i < len; ++i) {
+      address token = tokens[i];
+      if (token == _asset) {
+        tokenAmounts[i] = assetAmountForToken;
+      } else {
+        uint tokenBalance = _balance(token);
+
+        if (tokenBalance >= assetAmountForToken) { // we already have enough tokens
+          tokenAmounts[i] = tokenBalance;
+
+        } else { // we do not have enough tokens - borrow
+         _openPosition(_asset, assetAmountForToken - tokenBalance, token, ITetuConverter.ConversionMode.AUTO_0);
+          tokenAmounts[i] = _balance(token);
+        }
+      }
+    }
+
+    _depositorEnter(tokenAmounts);
+    _investedAssets += (assetBalanceBefore - _balance(_asset));
+  }
+
+  /// @dev Withdraw given amount from the pool.
+  function _withdrawFromPoolUniversal(uint amount, bool emergency) internal {
+    if (amount == 0) return;
+
+    if (!emergency) _doHardWork(false);
+
+    address _asset = asset;
+    uint assetBalanceBefore = _balance(_asset);
+
+    if (emergency) {
+      _depositorEmergencyExit();
+    } else {
+      uint liquidityAmount = amount * _depositorLiquidity() / _investedAssets;
+      _depositorExit(liquidityAmount);
```

Comment:
need to have a function fullExit() if requested amount >= invested balance

Reply:
I suggest require it from _depositorExit(amount) implementation - make full exit, when requested amount >= invested. 
Updated DepositorBase and MockDepositor.